src/Faker/Container/Container.php

```diff
@@ -87,35 +87,43 @@ private function getService(string $id, $definition)
     {
         if (is_callable($definition)) {
             try {
-                return $definition($this);
+                $service = $definition($this);
             } catch (\Throwable $e) {
                 throw new ContainerException(
                     sprintf('Error while invoking callable for "%s"', $id),
                     0,
                     $e,
                 );
             }
-        } elseif (is_object($definition)) {
+
+            return $service;
```

Comment:
Pulling out the `return` statement also allows to replace useless `elseif` statements with `if`.